src/prettytable/prettytable.py

```diff
@@ -1590,21 +1595,23 @@ def _compute_widths(self, rows, options) -> None:
                     widths[index] = max(widths[index], self.min_width[fieldname])
         self._widths = widths
 
+        per_col_padding = sum(self._get_padding_widths(options))
         # Are we exceeding max_table_width?
         if self._max_table_width:
             table_width = self._compute_table_width(options)
             if table_width > self._max_table_width:
                 # Shrink widths in proportion
-                scale = 1.0 * self._max_table_width / table_width
-                widths = [int(w * scale) for w in widths]
+                vrulers = len(widths) - 1
+                scale = (
+                    self._max_table_width - per_col_padding * len(widths) - vrulers
+                ) / (table_width - per_col_padding * len(widths) - vrulers)
+                widths = [max(1, int(w * scale)) for w in widths]
                 self._widths = widths
```

Comment:
We can combine these:
```suggestion
                self._widths = [max(1, int(w * scale)) for w in widths]
```